go/jsonclient/client.go

```diff
@@ -175,15 +181,16 @@ func (c *JSONClient) PostAndParseWithRetry(ctx context.Context, path string, req
 			// Request timeout, retry immediately
 		case httpRsp.StatusCode == http.StatusServiceUnavailable:
 			// Retry
-			backoffSeconds = backoffInterval
-			backoffInterval *= 2.0
-			if backoffInterval > maxInterval {
-				backoffInterval = maxInterval
-			}
+			backoffSeconds, backoffInterval = setBackoff(backoffInterval)
 			if retryAfter := httpRsp.Header.Get("Retry-After"); retryAfter != "" {
-				// TODO(drysdale): cope with a retry-after timestamp (RFC 7231 s7.1.3)
 				if seconds, err := strconv.Atoi(retryAfter); err == nil {
 					backoffSeconds = time.Duration(seconds) * time.Second
+				} else {
+					// Attempt to parse as a RFC1123 date
+					date, err := time.Parse(time.RFC1123, retryAfter)
+					if err == nil {
```

Comment:
More idiomatic to fold these two lines together; might also be possible to fold in with the `else` above too:

```go
if seconds, err := strconv.Atoi(retryAfter); err == nil {
	backoffSeconds = time.Duration(seconds) * time.Second
} else if date, err := time.Parse(time.RFC1123, retryAfter); err == nil {
	backoffSeconds = date.Sub(time.Now())
}
```